tests/test_koji_util.py

```diff
@@ -279,3 +281,121 @@ def test_tagging(self, task_state, failure):
         else:
             build_tag = tag_koji_build(session, build_id, target_name)
             assert build_tag == tag_name
+
+
+class TestGetKojiModuleBuild(object):
+    def mock_get_rpms(self, session):
+        (session
+            .should_receive('listArchives')
+            .with_args(buildID=1138198)
+            .once()
+            .and_return(
+                [{'btype': 'module',
+                  'build_id': 1138198,
+                  'id': 147879}]))
+        (session
+            .should_receive('listRPMs')
+            .with_args(imageID=147879)
+            .once()
+            .and_return([
+                {'arch': 'src',
+                 'epoch': None,
+                 'id': 15197182,
+                 'name': 'eog',
+                 'release': '1.module_2123+73a9ef6f',
+                 'version': '3.28.3'},
+                {'arch': 'x86_64',
+                 'epoch': None,
+                 'id': 15197187,
+                 'metadata_only': False,
+                 'name': 'eog',
+                 'release': '1.module_2123+73a9ef6f',
+                 'version': '3.28.3'},
+                {'arch': 'ppc64le',
+                 'epoch': None,
+                 'id': 15197188,
+                 'metadata_only': False,
+                 'name': 'eog',
+                 'release': '1.module_2123+73a9ef6f',
+                 'version': '3.28.3'},
+             ]))
+
+    def test_with_context(self):
+        module = 'eog:master:20180821163756:775baa8e'
+        koji_return = {
+            'build_id': 1138198,
+            'name': 'eog',
+            'version': 'master',
+            'release': '20180821163756.775baa8e',
+            'extra': {
+                'typeinfo': {
+                    'module': {
+                        'modulemd_str': 'document: modulemd\nversion: 2'
+                    }
+                }
+            }
+        }
+
+        spec = split_module_spec(module)
+        nvr = "{}-{}-{}.{}".format(spec.name, spec.stream, spec.version, spec.context)
```

Comment:
use a literal nvr here instead of calculating it, to reduce the chance of a gigo loop.

Reply:
Good suggestion - updated